src/main/java/com/redhat/composer/services/ChatBotService.java

```diff
@@ -88,19 +92,20 @@ public Multi<String> chat(ChatBotRequest request) {
 
     try {
       Multi<String> multi = Multi.createFrom().emitter(em -> {
-        List<Content> contentSources = new ArrayList<Content>();
         aiService.chatToken(request.getContext(), request.getMessage())
         .onNext(em::emit)
-        .onRetrieved(sources -> contentSources.addAll(sources))
+        .onRetrieved(sources -> {
+          try {
+            em.emit(objectMapper.writeValueAsString(new ContentResponse(sources)));
+            em.emit("\nEND_SOURCES_STRING\n");
```

Comment:
One small request can we do the same thing for before? So a "BEFORE_SOURCES_STRING"?

Reply:
Added "START_SOURCES_STRING" to show the start of the json payload